jerry-core/lit/lit-literal.h

```diff
@@ -16,42 +16,40 @@
 
-#include "lit-globals.h"
-#include "lit-literal-storage.h"
-#include "lit-magic-strings.h"
+#include "ecma-globals.h"
 
-void lit_init ();
-void lit_finalize ();
-void lit_dump_literals ();
+extern void lit_init ();
+extern void lit_finalize ();
+extern void lit_dump_literals ();
 
-literal_t lit_create_literal_from_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
-literal_t lit_find_literal_by_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
-literal_t lit_find_or_create_literal_from_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
+extern literal_t lit_create_literal_from_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
+extern literal_t lit_find_literal_by_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
+extern literal_t lit_find_or_create_literal_from_utf8_string (const lit_utf8_byte_t *, lit_utf8_size_t);
 
-literal_t lit_create_literal_from_num (ecma_number_t);
-literal_t lit_find_literal_by_num (ecma_number_t);
-literal_t lit_find_or_create_literal_from_num (ecma_number_t);
+extern literal_t lit_create_literal_from_num (ecma_number_t);
+extern literal_t lit_find_literal_by_num (ecma_number_t);
+extern literal_t lit_find_or_create_literal_from_num (ecma_number_t);
 
-bool lit_literal_equal_utf8 (literal_t, const lit_utf8_byte_t *, lit_utf8_size_t);
-bool lit_literal_equal_num (literal_t, ecma_number_t);
-bool lit_literal_equal (literal_t, literal_t);
+extern bool lit_literal_equal_utf8 (literal_t, const lit_utf8_byte_t *, lit_utf8_size_t);
+extern bool lit_literal_equal_num (literal_t, ecma_number_t);
+extern bool lit_literal_equal (literal_t, literal_t);
 
-bool lit_literal_equal_type_utf8 (literal_t, const lit_utf8_byte_t *, lit_utf8_size_t);
-bool lit_literal_equal_type_cstr (literal_t, const char *);
-bool lit_literal_equal_type_num (literal_t, ecma_number_t);
-bool lit_literal_equal_type (literal_t, literal_t);
+extern bool lit_literal_equal_type_utf8 (literal_t, const lit_utf8_byte_t *, lit_utf8_size_t);
+extern bool lit_literal_equal_type_cstr (literal_t, const char *);
+extern bool lit_literal_equal_type_num (literal_t, ecma_number_t);
+extern bool lit_literal_equal_type (literal_t, literal_t);
 
-const lit_utf8_byte_t *lit_literal_to_utf8_string (literal_t, lit_utf8_byte_t *, size_t);
-const char *lit_literal_to_str_internal_buf (literal_t);
+extern const lit_utf8_byte_t *lit_literal_to_utf8_string (literal_t, lit_utf8_byte_t *, size_t);
+extern const char *lit_literal_to_str_internal_buf (literal_t);
 
-literal_t lit_get_literal_by_cp (lit_cpointer_t);
+extern literal_t lit_get_literal_by_cp (lit_cpointer_t);
 
-lit_string_hash_t lit_charset_literal_get_hash (literal_t);
-ecma_number_t lit_charset_literal_get_number (literal_t);
-lit_utf8_size_t lit_charset_record_get_size (literal_t);
-ecma_length_t lit_charset_record_get_length (literal_t);
+extern lit_string_hash_t lit_charset_literal_get_hash (literal_t);
+extern ecma_number_t lit_charset_literal_get_number (literal_t);
```

Comment:
`lit_charset_literal_get_number` should be renamed to `lit_`**number**`_literal_get_number`, because charset literal could not hold a number.